jina/types/request/data.py

```diff
@@ -157,32 +168,33 @@ def proto_wo_data(
         calling :meth:`SerializeToString`.
         :return: protobuf instance containing parameters
         """
-        if self.is_decompressed:
-            return self._pb_body
-        elif self.is_decompressed_wo_data:
-            return self._pb_body_wo_data
-        else:
+        if self._pb_body is None:
             self._decompress_wo_data()
-            return self._pb_body_wo_data
+        return self._pb_body
```

Comment:
If `self._pb_body` is a proto *with* data (so not `None`), this will return a proto with data, right? That would be unexpected for me.

Reply:
Even if you originally wanted to only access only the parameters, if the proto is already loaded with data then you can just use it because the expensive step of deserialize the data is already done.

Reply:
I am guessing the user experience is exactly the same if a body with data is returned, but one without data is expected? If so, then it's ok